feat: Allow installing direct from git repository (publish.yaml‎ fix)

rjmunro opened the pull request:
## About the Contributor

This pull request is posted on behalf of SuperFlyTv


## Type of Contribution

This is a code improvement

## Current Behavior

If you try to use a forked version of this project in your package.json, for example by adding:

```json
		"atem-connection": "github:Sofie-Automation/sofie-atem-connection#rjmunro/install-from-git"
```

It would fail because it didn't know it had to build.

## New Behavior

It will build the package after installing from git.

I've modified the npm publish process a bit to 

## Testing Instructions

Create a trivial project in yarn and add this branch as a dependency:

```bash
yarn init
yarn add atem-connection@github:Sofie-Automation/sofie-atem-connection#rjmunro/install-from-git
```

Then check that it works. (TODO: Expand this)

## Other Information
<!-- The more information you can provide, the easier the pull request will be to merge -->


## Status
<!--
Before you open the PR, make sure the items below are done.
If they're not, please open the PR as a Draft.
-->

- [ ] PR is ready to be reviewed.
- [ ] The functionality has been tested by the author.
- [ ] Relevant unit tests has been added / updated.
- [ ] Relevant documentation (code comments, [system documentation](https://sofie-automation.github.io/sofie-core//)) has been added / updated.

rjmunro committed:
feat: Don't run build script on publish

It's already been run before. We do still need to run pinst, so we do
that manually.

Julusian commented on .github/workflows/publish.yaml:
```diff
@@ -172,8 +172,14 @@ jobs:
-          npm publish --access=public --provenance --tag ${{ needs.prepare.outputs.tag }}
+          npm publish --ignore-scripts --access=public --provenance --tag ${{ needs.prepare.outputs.tag }}
+
+          # Re-enable postinstall after publishing
```
there isn't much point in doing this re-enable, as the working directory is going to be thrown away after 3 more lines of this shell script

other than that, I dont have any problem with this

rjmunro commented:
@Julusian I've removed the re-enable and rebased this. The other PR already included some of this one, so this is now only the optimisation to the release process.
